dap_aria_mapping/pipeline/taxonomy_validation/distribution_statistics/tree_depth.py

```diff
@@ -0,0 +1,93 @@
+
+    Args:
+        tax (pd.DataFrame): taxonomy 
+        level (int): level of the taxonomy, first level is 0
+
+    Returns:
+        Tuple[int]: count of unique values
+    """
+    return tax.iloc[:, level].nunique()
+
+def _p_split(labels_at_level: set, labels_at_next_level: set) -> float:
+    """gets the fraction of topics at a given level that split into subtopics at the next level
+
+    Args:
+        labels_at_level (set): unique topics at a given level
+        labels_at_next_level (set): unique topics at the next level
+
+    Returns:
+        float: fraction of topics at a given level that split at the next level
+    """
+    return len([x for x in labels_at_level if x not in labels_at_next_level])/len(labels_at_level)
+
+def add_tax_metrics(tax: pd.DataFrame, tax_name: str, metrics: dict) -> dict:
+    """generates count of unique topics at each level, and percent of topics that split into subtopics at each level
+
+    Args:
+        tax (pd.DataFrame): taxonomy
+        tax_name (str): name of taxonomy
+        metrics (dict): dictionary to store metrics
+
+    Returns:
+        dict: dictionary with added metrics
+    """
+    for i, col in enumerate(tax.columns):
+        #add the counts of unique values at each level of the taxonomy to the metrics
+        #NOTE: given that the last level will carry down if a tree stops splitting, the bottom level contains some higer level groupings
+        metrics[tax_name]["counts"]['{}'.format(i+1)] = _count_at_level(tax, i)
+        #add the fraction of topics that split at a given level
+        if i < len(tax.columns)-1:
+            metrics[tax_name]["split_fractions"]['{}'.format(i+1)] = _p_split(set(tax.iloc[:, i]), set(tax.iloc[:, i+1]))
+
+    return metrics
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+            description="Taxonomy", formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+    parser.add_argument('--taxonomy', required = True,
+                    help='name of the taxonomy/ies to analyse. Available optons are: cooccur, centroids, imbalanced, a list of multiple, or all')
+    
+    args = parser.parse_args()
+
+    if args.taxonomy == 'all':
+        taxonomies = ['cooccur', 'centroids', 'imbalanced']
+    else:
+        taxonomies = list(args.taxonomy)
+    
+    metrics = {tax: {"counts": defaultdict(int), "split_fractions": defaultdict(int)} for tax in taxonomies}
+
+    if 'cooccur' in taxonomies:
+        cooccurence_taxonomy = get_cooccurrence_taxonomy()
+        metrics = add_tax_metrics(cooccurence_taxonomy, 'cooccur', metrics)
+    
+    if 'centroids' in taxonomies:
+        centroids_taxonomy = get_semantic_taxonomy(cluster_object='centroids').set_index('tag')
+        metrics = add_tax_metrics(centroids_taxonomy, 'centroids', metrics)
+    
+    if 'imbalanced' in taxonomies:
+        imbalanced_taxonomy = get_semantic_taxonomy(cluster_object='kmeans_strict_imb').set_index('tag')
```

Comment:
Old arguments for imbalanced, and no need for set_index in either centrids or imbalanced.